app/components/sy-topnav/component.js

```diff
@@ -12,4 +12,23 @@ export default class SyTopnav extends Component {
   get navMobile() {
     return this.media.isMo || this.media.isXs || this.media.isSm;
   }
+
+  /**
+   * TODO:
+   * i don't know if it's an issue in `ember-simple-auth` or `ember-simple-auth-oidc`
+   * or it's maybe wrong configration from Timed frontend side,
+   * BUT after refreshing the token, the session service is not encapsulating the reslts from keycloak
+   * inside ember data model, so i add this small.
+   *
+   * if the issue is fixed, change everything in app/abilities/report.js that belongs to `this.user?.id`
+   */
+  get fullName() {
+    const fullName = this.session.data.user?.fullName;
+    if (fullName) return fullName;
+
+    if (!this.session.data.user?.attributes) return "";
+
+    const userAttributes = this.session.data.user.attributes;
+    return `${userAttributes["first-name"]} ${userAttributes["last-name"]}`;
+  }
```

Comment:
I would first check if I really need this function. Otherwise you might be able to shorten it to:
```suggestion
  get fullName() {
    return this.session?.data?.user?.fullName ?? "${this.session?.data?.user?["firstName"]} ${this.session?.data?.user?["last-name"]}"
  }
```

Reply:
thats a lot of question marks :rofl:

Reply:
it was able for me to do that, but i wanted it to be more clear and readable :)